server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -268,7 +269,14 @@ synchronized void setMasterState(MasterState newState) {
     }
   }
 
-  private UpgradeCoordinator upgradeCoordinator;
+  private volatile UpgradeCoordinator upgradeCoordinator;
+
+  private UpgradeStatus getUpgradeStatus() {
+    if (upgradeCoordinator == null)
+      return UpgradeStatus.INITIAL;
+    return upgradeCoordinator.getStatus();
+  }
+
```

Comment:
Instead of relying on null, we should just set it to something that has a status of INITIAL instead. Then this variable can be final, and this method can be removed/inlined.

Reply:
That was a good idea.  That refactoring lead to some other slight refactorings and I like how the whole thing turned out.  Made the changes in d31fc6bf0a8b8887af735e26f6c4f9257af12eb9.